hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -378,6 +378,39 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		})
 	}
 
+	// Reconcile the HostedControlPlane audit webhook config if specified
+	// reference from the HostedCluster and syncing the secret in the control plane namespace.
+	{
+		if hcluster.Spec.AuditWebhook != nil && len(hcluster.Spec.AuditWebhook.Name) > 0 {
+			var src corev1.Secret
+			if err := r.Client.Get(ctx, ctrlclient.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.AuditWebhook.Name}, &src); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to get audit webhook config %s: %w", hcluster.Spec.AuditWebhook.Name, err)
+			}
+			configData, ok := src.Data[hyperv1.AuditWebhookKubeconfigKey]
+			if !ok {
+				return ctrl.Result{}, fmt.Errorf("audit webhook secret does not contain key %s", hyperv1.AuditWebhookKubeconfigKey)
+			}
+
+			hostedControlPlaneAuditWebhookSecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: controlPlaneNamespace.Name,
+					Name:      src.Name,
+				},
+			}
+			_, err = controllerutil.CreateOrUpdate(ctx, r.Client, hostedControlPlaneAuditWebhookSecret, func() error {
+				if hostedControlPlaneAuditWebhookSecret.Data == nil {
+					hostedControlPlaneAuditWebhookSecret.Data = map[string][]byte{}
+				}
+				hostedControlPlaneAuditWebhookSecret.Data[hyperv1.AuditWebhookKubeconfigKey] = configData
+				hostedControlPlaneAuditWebhookSecret.Type = corev1.SecretTypeOpaque
+				return nil
+			})
+			if err != nil {
+				r.Log.Error(err, "Error reconciling audit webhook secret. Continuing.", "namespace", controlPlaneNamespace.Name)
```

Comment:
why isn't this returning the error but just logging?

Reply:
Agree, need to return early here on error. Can also refactor to the inline `if err := ...` style at the same time for consistency with other blocks

Reply:
So this was based off a previous iteration that I think I'm trying to stay close too but I agree.

Reply:
Copy pasta problem where I was trying to stay consistent with something that didn't match changing